### PR TITLE
Coverity fixes for ssl.c

### PR DESCRIPTION
**Pull request by embhorn**
CID list: 184992 185005 185042 185091 185100 185105 185117 185124 185125 185149 185154 185160

### CODE REVIEW

**Comment by embhorn** [comment]
I'm trying to break the review up by making commits for one file at a time. @dgarske, I am defaulting to you as a reviewer, but if you think someone else would have valuable insight for these changes please reassign the reviewer role.

**Comment by embhorn** [comment]
I guess I should have waited to add you as a reviewer until all the commits are in place. My plan was to do a review for each file, but the only one with significant changes was ssl.c. I should wrap up the changes for the wolfssl/src dir today. I will re-start the review afterwards.

File: src/bio.c [hunk 1]
```diff
@@ -134,7 +134,7 @@ static int wolfSSL_BIO_SSL_read(WOLFSSL_BIO* bio, void* buf,
     WOLFSSL_ENTER("wolfSSL_BIO_SSL_write");
```
Comment (dgarske) [inline src/bio.c]: Can you change this WOLFSSL_ENTER  message to `wolfSSL_BIO_SSL_read`?

File: src/bio.c [hunk 2]
```diff
@@ -134,7 +134,7 @@ static int wolfSSL_BIO_SSL_read(WOLFSSL_BIO* bio, void* buf,
...
 
     /* already got eof, again is error */
-    if (bio && front->eof)
+    if (front->eof)
```
Comment (dgarske) [inline src/bio.c]: This probably should be `if (front && front->eof)`.

File: src/bio.c (hunk 1 above)
Comment (embhorn) [reply src/bio.c]: Yes.

File: src/bio.c (hunk 2 above)
Comment (embhorn) [reply src/bio.c]: Is the intention to avoid dereferencing "front" when it's NULL? Since this condition returns an error when true, it could be:
```    
if ((front == NULL) || front->eof)
        return WOLFSSL_FATAL_ERROR;
```

Comment (dgarske) [reply src/bio.c]: This logic is hard to follow, but the `wolfSSL_BIO_read` API is an openssl compatibility one. If in doubt about functionality sometimes its helpful to go check the openssl sources. In this case the NULL check on front is to prevent a NULL dereference. If the front->eof is set then the read has reached EOF and it should not try and read again.
